include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -149,14 +149,14 @@ def ProcedureOp : MooreOp<"procedure", [
 def VariableOp : MooreOp<"variable", [
   DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
   OptionalTypesMatchWith<"initial value and variable types match",
-    "result", "initial", "$_self">,
+    "result", "initial", "cast<RefType>($_self).getNestedType()">
 ]> {
   let summary = "A variable declaration";
   let description = [{
     See IEEE 1800-2017 § 6.8 "Variable declarations".
   }];
   let arguments = (ins StrAttr:$name, Optional<UnpackedType>:$initial);
-  let results = (outs Res<UnpackedType, "", [MemAlloc]>:$result);
+  let results = (outs Res<RefType, "", [MemAlloc]>:$result);
   let assemblyFormat = [{
     custom<ImplicitSSAName>($name) ($initial^)? attr-dict
```

Comment:
I think `` is ignored here. But #7112 complements it.

Reply:
The `` prevents an additional space to be printed in case there is no name:
```
moore.variable  %0 : i32  // without ``
moore.variable %0 : i32   // with ``
```

Reply:
You're totally right! I remember you mentioned this difference in another PR.